perception/navigator_vision/nodes/database_color.py

```diff
@@ -306,70 +314,70 @@ def _get_closest_color(self, hue_angle):
             cg = np.cos(h_val)
             sg = np.sin(h_val)
             # We need a signed error for some math later on so no absolute value
-            this_error = np.arctan2(sg*c - cg*s, cg*c + sg*s)
+            this_error = np.arctan2(sg * c - cg * s, cg * c + sg * s)
             if np.abs(this_error) < np.abs(error):
                 error = this_error
                 likely_color = color
 
         fprint("Likely color: {} with an hue error of {} rads.".format(likely_color, np.round(error, 3)))
-        return [likely_color, error] 
+        return [likely_color, error]
 
     def do_estimate(self, totem_id):
         """Calculates best color estimate for a given totem id"""
-        fprint("DOING ESTIMATE", msg_color='blue') 
+        fprint("DOING ESTIMATE", msg_color='blue')
         if totem_id not in self.colored:
             fprint("Totem ID {} not found!".format(totem_id), msg_color='red')
             return None
-       
+
         t_color = self.colored[totem_id]
-        
+
         if len(t_color) < self.min_obs:
             fprint("Only {} observations. {} required.".format(len(t_color), self.min_obs), msg_color='red')
             return None
 
-        kwargs = {'v_u': self.v_u, 'v_sig': self.v_sig, 'dist_sig': self.dist_sig, 
+        kwargs = {'v_u': self.v_u, 'v_sig': self.v_sig, 'dist_sig': self.dist_sig,
                   'q_factor': self.q_factor, 'q_sig': self.q_sig}
 
         w, weights = t_color.compute_confidence([self.v_factor, self.dist_factor, self.q_factor], True, **kwargs)
         fprint("CONF: {}".format(w))
         if np.mean(w) < self.conf_reject:
             return None
-        
-        hue_angles = np.radians(np.array(t_color.hues) * 2) 
+
+        hue_angles = np.radians(np.array(t_color.hues) * 2)
         angle = self._compute_average_angle(hue_angles, w)
         color = self._get_closest_color(angle)
-        
-        msg = t_color.as_message 
+
+        msg = t_color.as_message
         msg.id = totem_id
         msg.confidence = w
         msg.labels = ["value_errs", "dists", "q_diffs"]
         msg.weights = weights
-        msg.color = colors[0]
+        msg.color = color[0]
```

Comment:
nice

Reply:
This script has so many bugs I have no idea why it's in the repo lol